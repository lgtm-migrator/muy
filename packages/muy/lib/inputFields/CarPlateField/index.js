import React from "react"
import TextField from "@material-ui/core/TextField"
import makeStyles from "@material-ui/core/styles/makeStyles"
import propTypes from "../propTypes"

const useStyles = makeStyles({
  input: {
    textTransform: "uppercase",
  },
})

const CarPlateField = ({
  autoCapitalize,
  id,
  inputMode,
  inputProps,
  maxLength,
  minLength,
  name,
  pattern,
  required,
  ...props
}) => {
  const classes = useStyles()
  return (
    <TextField
      autoComplete
      id={id}
      inputProps={{
        autoCapitalize,
        "aria-required": required,
        "data-name": name,
        inputMode,
        maxLength,
        minLength,
        pattern,
        ...inputProps,
      }}
      InputProps={{
        classes: classes,
      }}
      name={name}
      required={required}
      {...props}
    />
  )
}

export default CarPlateField

CarPlateField.defaultProps = {
  autoCapitalize: "characters",
  autoComplete: "off",
  autoCorrect: "off",
  id: "carPlate",
  inputMode: "verbatim",
  maxLength: 7,
  minLength: 7,
  name: "carPlate",
  pattern: "[a-zA-Z]{3}[0-9]([a-zA-Z]|[0-9])[0-9]{2}",
}

CarPlateField.propTypes = {
  autoCapitalize: propTypes.autoCapitalize,
  autoComplete: propTypes.autoComplete,
  id: propTypes.id,
  inputMode: propTypes.inputMode,
  inputProps: propTypes.inputProps,
  name: propTypes.name,
  required: propTypes.required,
}
