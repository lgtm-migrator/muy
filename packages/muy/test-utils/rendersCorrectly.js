import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
