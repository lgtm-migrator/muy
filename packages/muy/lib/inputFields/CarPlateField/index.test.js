import React from "react"
import rendersCorrectly from "../../../test-utils/rendersCorrectly"
import CarPlateField from "."

describe("<CarPlateField /> field", () => {
  it("should render correctly", () => {
    rendersCorrectly(<CarPlateField />)
  })
})
