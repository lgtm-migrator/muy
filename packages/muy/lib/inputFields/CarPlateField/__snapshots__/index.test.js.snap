// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarPlateField /> field should render correctly 1`] = `
<div
  autoCorrect="off"
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="characters"
      autoComplete="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input makeStyles-input-1"
      data-name="carPlate"
      disabled={false}
      id="carPlate"
      inputMode="verbatim"
      maxLength={7}
      minLength={7}
      name="carPlate"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      pattern="[a-zA-Z]{3}[0-9]([a-zA-Z]|[0-9])[0-9]{2}"
      required={false}
      type="text"
    />
  </div>
</div>
`;
